app/models/message.rb

```diff
@@ -56,7 +59,17 @@ def chat_message_link
     Rails.application.routes.url_helpers.contributor_request_path(
       contributor,
       request,
-      anchor: "chat-row-#{id}"
+      anchor: "message-#{id}"
     )
   end
+
+  private
+
+  def notify_recipient
+    if reply?
+      MessageReceived.with(contributor: sender, request: request, message: self).deliver_later(User.all)
```

Comment:
`deliver_later` enqueues a job. Jobs are written to the database. Once the job is executed, it will then delivery the notification, which in our case means writing a notification record to the database.

So maybe we could just use `deliver` instead of `deliver_later` here? Writing a job to the database isn’t going to be faster than writing the actual notification to the database.

Reply:
Do you mean just for the `MessageRecieved` type?

I guess the benefit of enqueuing a job is not that it's faster, but that it happens in the background and the user does not need to wait for it to happen. In our case, the `MessageReceived` type wouldn't need to concern itself with this since there is no user interaction with the site, but for `OnboardingCompleted` and `ChatMessageSent` there would be. Or are you just not concerned with the time it takes to create a record for each User, which is typically quite few?

Reply:
Note that making this change resulted in many errors popping up that were resolved when removing, rebuilding, and deploying an image without this proposed change. See [monitoring channel discussion](https://tactilenews.slack.com/archives/C01ACFHFTV2/p1662381100797989).

It doesn't make sense to me how this would result in a `ActiveModel::UnknownAttributeError` for `contributor_id`, but I tested the error/fix flow on multiple instances, staging and demo and reliably experienced the error/saw the fix get rid of the error. When I manually created the notifications in the rails console I did not get an error.